lib/LINZ/Bde.pm

```diff
@@ -477,6 +419,137 @@ sub copy
     return $result;
 }
 
+sub copy
+{
+    my($self,$outputfile,@options) = @_;
+
+    my $exe = _bdecopy();
+    my $opts = ref($options[0]) ? $options[0] : {@options};
+    my @copyopts;
+    my $cfgtmp;
+    my $cfg = $opts->{config};
+    if( $cfg =~ /\n/)
+    {
+        use File::Temp;
+        my $htmp;
+        ($htmp,$cfgtmp) = File::Temp::tempfile();
+        print $htmp $cfg;
+        CORE::close($htmp);
+        $cfg = $cfgtmp;
+    }
+    push(@copyopts,'-c',$cfg) if $cfg;
+
+    my @addfiles;
+    push(@addfiles,$opts->{addfile}) if $opts->{addfile};
+    push(@addfiles,@{$self->{archive_files}}) 
+        if $self->{archive_files} && $opts->{use_archive};
+    my $addfile = join('+',@addfiles);
+    push(@copyopts,'-p',$addfile) if $addfile;
+
+    push(@copyopts,'-a') if $opts->{append};
+    push(@copyopts,'-z') if $opts->{compress};
+
+    my $flds = join(':',split(' ',$self->{override_flds}));
+    push(@copyopts,'-f',$flds) if $flds;
+
+
+    $flds = $opts->{output_fields};
+    if( $flds )
+    {
+        if( ! ref($flds) )
+        {
+            my @f = $flds =~ /\w+/g;
+            $flds = \@f;
+        }
+        $self->output_fields($flds);
+    }
+
+    if( $self->{output_fields} )
+    {
+        $flds = join(':',@{$self->{output_fields}});
+        push(@copyopts,'-o',$flds) if $flds;
+    }
+
+    my $log = $opts->{log_file} || $outputfile.".log";
+
+    system(
+        $exe,
+        @copyopts,
+        $self->{path},
+        $outputfile,
+        $log
+    );
+    unlink($cfgtmp) if $cfgtmp;
+
+    my $result = $self->resultFromLog($log);
+    unlink($log) if $opts->{log_file} || ! $opts->{keep_log};
+    return $result;
+}
+
+sub pipe
```

Comment:
New function seems good. But before you merge I suggest refactoring the code between the copy and pipe methods as they almost have the same code.